packages/http/src/validator/index.ts

```diff
@@ -143,8 +144,21 @@ const validateOutput: ValidatorFn<IHttpOperation, IHttpResponse> = ({ resource,
   );
 };
 
-function getPathParams(path: string, template: string) {
-  return new URI.Template(template).match(path);
+function getPathParams(path: string, template: string, realParams: IHttpPathParam[]) {
+  // replace "-" in template path and query params
+  const cleanedTemplate = template.replace(/(?<=\{.*)(?=.*\})(-)/g, '');
+
+  const pathParams = new URI.Template(cleanedTemplate).match(path);
+
+  // get pathParams back into real format (i.e put "-" back)
+  const finalPathParams = {};
+  for (const realParam of realParams) {
+    const cleanedName = realParam.name.replace(/-/g, '');
+    if (pathParams[cleanedName]) {
+      finalPathParams[realParam.name] = pathParams[cleanedName];
+    }
+  }
+  return finalPathParams;
```

Comment:
This function is fundamentally ignoring the parameter in case it cannot be found in the "real" list. I think we should communicate we haven't found a parameter to the caller, so I think we need to employ the `Option` monad here, and then the caller will decide what to do?

Reply:
well the real list contains params from the spec itself. So if it's not in there, of course it won't be added. I don't really see an issue with having this being driven from the params that are actually in the spec.

But, if we can't find a "real" spec param in the list of params generated by `uri-template-lite` after cleaning it's name the same way we did to build the list of params with `uri-template-lite` then we should probably throw an error saying something like `Error getting path param {real_param_name} from the path provided`.

So I don't know if we really need to do what you're proposing with the Option monad (but to be honest I don't really know what you mean by Option monad 😬 )